src/Lookup/InMemoryEntityLookup.php

```diff
@@ -18,6 +21,7 @@
 class InMemoryEntityLookup implements EntityLookup {
 
 	private $entities = array();
+	private $exceptions = array();
```

Comment:
I know what you mean, but I still disagree. Especially here where the type `EntityLookupException[]` in nowhere else in this file. What this array contains only becomes clear when you search the code for all places that add stuff to this array. You need to read the **whole** file, not just a constructor or something.